pkgs/shells/bash/ble.sh/default.nix

```diff
@@ -0,0 +1,66 @@
+{ lib, stdenvNoCC, fetchgit, git, curl, jq, nix-prefetch, runtimeShell }:
+
+stdenvNoCC.mkDerivation {
+  name = "ble.sh";
+  version = "0.3.3";
+
+  src = fetchgit {
+    url = "https://github.com/akinomyoga/ble.sh.git";
+    rev = "refs/tags/v0.3.3";
+    hash = "sha256-Gfo2S1t5Kdy+8TEDS4M5yhyRShvzQIljdE0MQK1CL+4=";
+    fetchSubmodules = true;
+    leaveDotGit = true;
+  };
+
+  nativeBuildInputs = [ git ];
+  buildInputs = [ curl jq nix-prefetch ];
+
+  doCheck = true;
+
+  installFlags = [ "INSDIR=$(out)/share" ];
+  postInstall = ''
+    mkdir -p "$out/bin"
+    cat <<SCRIPT >"$out/bin/ble.sh-share"
+    #!${runtimeShell}
+    # Run this script to find the ble.sh shared folder
+    # where all the shell scripts are living.
+    echo "$out/share/ble.sh"
+    SCRIPT
+    chmod +x "$out/bin/ble.sh-share"
+
+    mkdir -p "$out/share/lib"
+    cat <<SCRIPT >"$out/share/lib/_package.sh"
+    _ble_base_package_type=nix
+
+    function ble/base/package:nix/update {
+      local rev=\$(${curl}/bin/curl -sL 'https://api.github.com/repos/akinomyoga/ble.sh/branches/master' | ${jq}/bin/jq -r '.commit.sha')
+      local hash="\$(${nix-prefetch}/bin/nix-prefetch fetchgit --url 'https://github.com/akinomyoga/ble.sh.git' --branchName master --rev \$rev --leaveDotGit --fetchSubmodules 2>/dev/null)"
+
+      cat <<-EOS >/dev/stderr
+    	You cannot update ble.sh automatically via Nix.
+    	The latest ble.sh from master branch can be installed by overriding nixpkgs like following.
+
+    	  pkgs.ble-sh.overrideAttrs (_: {
+    	    src = pkgs.fetchgit {
+    	      url = "https://github.com/akinomyoga/ble.sh.git";
+    	      rev = "\$rev";
+    	      hash = "\$hash";
+    	      fetchSubmodules = true;
+    	      leaveDotGit = true;
+    	    };
+    	    doCheck = false;
+    	  });
+    	EOS
+      return 1
+    }
+    SCRIPT
```

Comment:
```suggestion
```
just package the latest commit from master. The auto updater should also work with that out of the box.

Reply:
Can you take a look at a discussion I mentioned above? It's already discussed.

Reply:
Oh wait, by "auto updater", do you mean that updating nixpkgs should also update ble.sh automatically?
I thought you're talking about ble-update because you mentioned to `_package.sh`!

Could you explain why you think you should omit this part, with how you think it is related to nix's auto updater?
I don't understand because it seems unrelated. Nix package is highly possible to be behind ble.sh's master branch, and ble-update is just kind enough to tell users how to use the latest version when they want to update it eagerly.

Again, this is already discussed at https://github.com/akinomyoga/ble.sh/discussions/212#discussioncomment-3196490. `_package.sh` is to disable `ble-update` which can break nix store.

Reply:
```
    	You cannot update ble.sh automatically via Nix.
```

I think this line is the source of confusion. What does it actually mean? I think it is actually kind of opposite, i.e., "You cannot automatically update ble.sh using ble-update. You can update ble.sh via Nix."

Reply:
What I wanted to say is "You are unable to update ble.sh using `ble-update` command if you install it via Nix. And it is also impossible to install the latest master branch without configuration".

I'll update that explanation to avoid confusion. Thanks👍

Reply:
I remembered that I already updated it on d3e6b966deae0167bd4080ccc0a78854d7be0210.
This is the current description:

> You cannot update ble.sh to the latest version with Nix.
> The latest ble.sh from master branch can be installed by overriding nixpkgs like following.

When this description is shown, the current version of ble.sh is different from master branch, which means that ble.sh from nixpkgs is some commits behind. So the user is unable to update it using Nix.

How about this one? Still confusing?

Reply:
> When this description is shown, the current version of ble.sh is different from master branch,

This message would be shown any time the user calls `ble-update` regardless of whether the current version matches with the master or not.

In addition, `ble-update` is not the command to update ble.sh to the master version, but the command to update it to the latest version of the current installation source. For example, if you are using v0.3, ble-update will update ble.sh to the latest version of `v0.3-master`. If you are using the nightly version, it will update ble.sh to the latest version of `ble-nightly.tar.xz` distributed on the release page. If you are using `blesh` in AUR, it will update ble.sh to the latest version of ble-0.3 in AUR. `blesh-git` in AUR is exceptional but intensionally designed to use the latest `master`.

Oh, in this sense, maybe we can make `ble-update` update ble.sh to the latest version in `nixpkgs` when the local version is behind the version in nixpkgs? Actually, this is the expected behavior of `ble-update`.

Reply:
> This message would be shown any time the user calls ble-update regardless of whether the current version matches with the master or not.

I inserted version-recognition script so that it should return 6 if no update is needed.

https://github.com/NixOS/nixpkgs/blob/d3e6b966deae0167bd4080ccc0a78854d7be0210/pkgs/shells/bash/ble.sh/default.nix#L39

> we can make ble-update update ble.sh to the latest version in nixpkgs

This seems nicer! Executing `nix-env -iA blesh` may be enough. Simpler solution.

Reply:
> I inserted version-recognition script so that it should return 6 if no update is needed. That's why the users won't see the message when they are using the latest.

Ah, OK! I see. You are right!

Reply:
Fixed